bundles/org.openhab.binding.surepetcare/src/main/java/org/openhab/binding/surepetcare/internal/SurePetcareAPIHelper.java

```diff
@@ -0,0 +1,491 @@
+        JsonParser parser = new JsonParser();
+        JsonObject object = (JsonObject) parser.parse(apiResult);
+        return object.get("data");
+    }
+
+    /**
+     * Sends a given object as a JSON payload to the API.
+     *
+     * @param url           the URL
+     * @param requestMethod the request method (POST, PUT etc.)
+     * @param payload       an object used for the payload
+     * @throws SurePetcareApiException
+     */
+    private void setDataThroughApi(String url, HttpMethod method, Object payload) throws SurePetcareApiException {
+        String jsonPayload = SurePetcareConstants.GSON.toJson(payload);
+        postDataThroughAPI(url, method, jsonPayload);
+    }
+
+    /**
+     * Returns the result of a GET API call as a string.
+     *
+     * @param url the URL
+     * @return a JSON string with the API result
+     * @throws SurePetcareApiException
+     */
+    private String getResultFromApi(String url) throws SurePetcareApiException {
+        boolean success = false;
+        String responseData = "";
+        while (!success) {
+            try {
+                Request request = httpClient.newRequest(url).method(HttpMethod.GET);
+                setConnectionHeaders(request);
+                ContentResponse response = request.send();
+                if (response.getStatus() == HttpURLConnection.HTTP_OK) {
+                    responseData = response.getContentAsString();
+                    logger.debug("API execution successful, response: {}", responseData);
+                    success = true;
+                } else {
+                    logger.debug("HTTP Response Code: {}", response.getStatus());
+                    logger.debug("HTTP Response Msg: {}", response.getReason());
+                    if (response.getStatus() == HttpURLConnection.HTTP_UNAUTHORIZED) {
+                        // authentication token has expired, login again and retry
+                        login(username, password);
+                    } else {
+                        throw new SurePetcareApiException(
+                                "Http error: " + response.getStatus() + " - " + response.getReason());
+                    }
+                }
+            } catch (AuthenticationException | InterruptedException | ExecutionException | TimeoutException
+                    | ProtocolException e) {
+                logger.debug("Exception caught during API execution: {}", e.getMessage());
+                throw new SurePetcareApiException(e);
+            }
+        }
+        return responseData;
+    }
+
+    /**
+     * Uses the given request method to send a JSON string to an API.
+     *
+     * @param url         the URL
+     * @param method      the required request method (POST, PUT etc.)
+     * @param jsonPayload the JSON string
+     * @throws SurePetcareApiException
+     */
+    private void postDataThroughAPI(String url, HttpMethod method, String jsonPayload) throws SurePetcareApiException {
```

Comment:
Can you refactor this method and `getResultFromApi` into a single method?

Reply:
I've refactored the 2 methods, so that they both use the same code for the http API execution.